homeassistant/components/pyload/config_flow.py

```diff
@@ -0,0 +1,122 @@
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Required(CONF_SSL, default=False): cv.boolean,
+        vol.Required(CONF_VERIFY_SSL, default=True): bool,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, user_input: dict[str, Any]) -> None:
+    """Validate the user input and try to connect to PyLoad."""
+
+    session = async_create_clientsession(
+        hass,
+        user_input[CONF_VERIFY_SSL],
+        cookie_jar=CookieJar(unsafe=True),
+    )
+
+    url = (
+        f"{"https" if user_input[CONF_SSL] else "http"}://"
+        f"{user_input[CONF_HOST]}:{user_input[CONF_PORT]}/"
+    )
+    pyload = PyLoadAPI(
+        session,
+        api_url=url,
+        username=user_input[CONF_USERNAME],
+        password=user_input[CONF_PASSWORD],
+    )
+
+    await pyload.login()
+
+
+class PyLoadConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for pyLoad."""
+
+    VERSION = 1
+    # store values from yaml import so we can use them as
+    # suggested values when the configuration step is resumed
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {CONF_HOST: user_input[CONF_HOST], CONF_PORT: user_input[CONF_PORT]}
+            )
+            try:
+                await validate_input(self.hass, user_input)
+            except (CannotConnect, ParserError):
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                title = user_input.pop(CONF_NAME, None)
+                return self.async_create_entry(
+                    title=title or DEFAULT_NAME, data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                STEP_USER_DATA_SCHEMA, user_input
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_info: dict[str, Any]) -> ConfigFlowResult:
+        """Import config from yaml."""
+
+        config = {
+            CONF_NAME: import_info.get(CONF_NAME),
+            CONF_HOST: import_info.get(CONF_HOST, DEFAULT_HOST),
+            CONF_PASSWORD: import_info.get(CONF_PASSWORD, ""),
```

Comment:
Why is the password required btw

Reply:
> Why is the password required btw

Optional you mean?

Reply:
not sure what you mean?

Reply:
It previously wasn't required, why is it now

Reply:
in the yaml configuration all fields were marked as optional, what i really don't understand because you cannot connect without username and password. For the other fields were default values, but not for these.